### ISSUE TITLE
Assertion failed: (getOperand(0)->getType() == cast<PointerType>(getOperand(1)->getType())->getElementType() && "Ptr must be a pointer to Val type!"), function AssertOK, file Instructions.cpp, line 1083.

### ISSUE THREAD

**Issue by timotheecour** [ldc-developers/ldc#965]
ldc2 -c -od=/tmp/d08 main.d
//main.d:

```d
class A{}
void fun() {
    A a;
    auto b=a?typeid(a):typeid(a);
}
```

Assertion failed: (getOperand(0)->getType() == cast<PointerType>(getOperand(1)->getType())->getElementType() && "Ptr must be a pointer to Val type!"), function AssertOK, file Instructions.cpp, line 1083.
0  ldc2                     0x0000000108e35ad1 llvm::sys::PrintStackTrace(__sFILE*) + 57
1  ldc2                     0x0000000108e36027 SignalHandler(int) + 217
2  libsystem_platform.dylib 0x00007fff9a658f1a _sigtramp + 26
3  ldc2                     0x00000001091cbc27 nuls + 69727
4  ldc2                     0x0000000108e35eb6 abort + 22
5  ldc2                     0x0000000108e35ea0 abort + 0
6  ldc2                     0x0000000108dcc43b llvm::StoreInst::AssertOK() + 211
7  ldc2                     0x000000010853df39 DtoStore(llvm::Value*, llvm::Value*) + 105
8  ldc2                     0x00000001085396cd ToElemVisitor::visit(CondExp*) + 749
9  ldc2                     0x00000001085357cc ToElemVisitor::visit(AssignExp*) + 1164
10 ldc2                     0x000000010852cfaa toElem(Expression*) + 58
11 ldc2                     0x00000001084f9edf DtoVarDeclaration(VarDeclaration*) + 815
12 ldc2                     0x00000001084fa0d9 DtoDeclarationExp(Dsymbol*) + 313
13 ldc2                     0x0000000108530db7 ToElemVisitor::visit(DeclarationExp*) + 87
14 ldc2                     0x000000010852d226 toElemDtor(Expression*) + 614
15 ldc2                     0x000000010851c4f2 ToIRVisitor::visit(ExpStatement*) + 130
16 ldc2                     0x000000010851c584 ToIRVisitor::visit(CompoundStatement*) + 116

ldc2 --version

LDC - the LLVM D compiler (9d3dd1):
  based on DMD v2.066.1 and LLVM 3.6.1
  Default target: x86_64-apple-darwin14.3.0
  Host CPU: core-avx2

**Comment by redstar** [comment]
@kinke already fixed this bug in the merge-2.067 branch: https://github.com/ldc-developers/ldc/commit/9ecc483b14a39cad377af0ea84d4d05db826d94e.
I cherry-picked this commit to master branch and added your example as test case.

**Comment by timotheecour** [comment]
thanks!
